backend/inspirehep/factory.py

```diff
@@ -22,6 +23,37 @@ def config_loader(app, **kwargs_config):
     return result
 
 
+class DispatcherMiddlewareWithMatchingHostNameAndServerName(DispatcherMiddleware):
+    """WSGI middleware without checking hostname.
+
+    It will override the request ``HOST`` to always match
+    the ``SERVER_NAME``.
+    """
+
+    def __init__(self, app, mounts=None, server_name=None):
+        self.app = app
+        self.mounts = mounts or {}
+        self.server_name = server_name
+        super().__init__(app, mounts)
+
+    def __call__(self, environ, start_response):
+        if self.server_name:
+            environ["HTTP_HOST"] = self.server_name
```

Comment:
so taking it from the `environ['SERVER_NAME']` is not OK in the end?

Reply:
I was wrong, actually if ```HTTP_HOST``` is present (https://github.com/pallets/werkzeug/blob/b81aa0f18c7336a0421b569d8a5dab8d5c0d7517/src/werkzeug/wsgi.py#L161 and https://github.com/pallets/werkzeug/blob/master/src/werkzeug/routing.py#L1510) is been checked against the wsgi_app ``SERVER_NAME`` not the ``environ['SERVER_NAME']`` because the app ``SERVER_NAME`` is not ``None`` https://github.com/pallets/werkzeug/blob/master/src/werkzeug/routing.py#L1513-L1516

Reply:
I can saw you IRL